src/directives/scaleselector.js

```diff
@@ -111,14 +111,18 @@ ngeo.ScaleselectorController = function($scope, $element, $attrs) {
       ($scope.$eval(scalesExpr));
   goog.asserts.assert(this.scales !== undefined);
 
-  var zoomLevels = Object.keys(this.scales).map(Number);
-  zoomLevels.sort(ol.array.numberSafeCompareFunction);
-
   /**
    * @type {Array.<number>}
    * @export
    */
-  this.zoomLevels = zoomLevels;
+  this.zoomLevels;
+
+  $scope.$watchCollection(goog.bind(function() {
```

Comment:
Ok, so a slight improvement could be to watch the `Object.keys(this.scales).length` and use a simple `$watch`.
`$watchCollection` watch all properties of your object on all digest loops, can sometimes be avoided.

Could you check you have same result ?